hatchways/team-bluejay
# Kareem jwt cookie authentication #15 #16

Ticekts: #15 , #16 

Description:
- Added jwt via cookies support for the backend API
- Added refresh token support for backend API
- Added csrf token support
- Connected the front end with the backend for login and sign up
- Added a boostrapper component to load the user from the backend on application startup
- added alert context for raising errors throughout the application
- added a loggout route to the backend
- added a protected get logged in user route to the backend 
- added a token refresh route to the backend (not used by the front end yet)

Testing
1. Pull/install/build 
2. Navigate to the / path in the website
3. verify that you got redirected to login page
4. sign up with a new user
5. verify that you got logged in after sign up and the user name is displayed in the navbar/appbar
6. Refresh the page or navigate to /login or /signup
7. verify that you are still logged in and navigated to the main page /
8. click logout
9. verify that you got redirected to the login page
10. attempt to navigate to the hompage again, and verify that you cant access it
11. enter an incorrect password
12. verify that a snack appears showing the error from the backend
